Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -183,15 +189,29 @@ std::array<int, 3> HGCalDDDConstants::assignCellTrap(float x, float y, float z,
   if (indx.first < 0)
     return std::array<int, 3>{{irad, iphi, type}};
   double xx = (z > 0) ? x : -x;
-  double r = (reco ? std::sqrt(x * x + y * y) : HGCalParameters::k_ScaleFromDDD * std::sqrt(x * x + y * y));
-  double phi = (r == 0. ? 0. : std::atan2(y, xx));
+  double phi = (((y == 0.0) && (x == 0.0)) ? 0. : std::atan2(y, xx));
   if (phi < 0)
     phi += (2.0 * M_PI);
+  iphi = 1 + (int)(phi / indx.second);
```

Comment:
`indx.second` is explicitly allowed to be 0. There should be protection against division by 0.
```suggestion
if (indx.second != 0)
  iphi = 1 + (int)(phi / indx.second);
```

Reply:
Trying to take care of this